Flaky test_premium::test_upload_data_to_server

## Problem Definition

Fails some times in Github actions and I think I have also seen it happen locally.
Link here: https://github.com/rotki/rotki/pull/854/checks?check_run_id=569012715

```
2020-04-07T21:57:03.9119869Z =================================== FAILURES ===================================
2020-04-07T21:57:03.9125077Z _______________________ test_upload_data_to_server[True] _______________________
2020-04-07T21:57:03.9125463Z 
2020-04-07T21:57:03.9125858Z rotkehlchen_instance = <rotkehlchen.rotkehlchen.Rotkehlchen object at 0x7f4f1740be10>
2020-04-07T21:57:03.9127656Z username = 'testuser', db_password = '123'
2020-04-07T21:57:03.9128310Z 
2020-04-07T21:57:03.9129545Z     @pytest.mark.parametrize('start_with_valid_premium', [True])
2020-04-07T21:57:03.9129748Z     def test_upload_data_to_server(rotkehlchen_instance, username, db_password):
2020-04-07T21:57:03.9129906Z         """Test our side of uploading data to the server"""
2020-04-07T21:57:03.9130088Z         last_ts = rotkehlchen_instance.data.db.get_last_data_upload_ts()
2020-04-07T21:57:03.9130866Z         assert last_ts == 0
2020-04-07T21:57:03.9130998Z     
2020-04-07T21:57:03.9133370Z         # Write anything in the DB to set a non-zero last_write_ts
2020-04-07T21:57:03.9134973Z         rotkehlchen_instance.data.db.set_settings(ModifiableDBSettings(main_currency=A_GBP))
2020-04-07T21:57:03.9135546Z         last_write_ts = rotkehlchen_instance.data.db.get_last_write_ts()
2020-04-07T21:57:03.9135985Z         _, our_hash = rotkehlchen_instance.data.compress_and_encrypt_db(db_password)
2020-04-07T21:57:03.9136583Z         remote_hash = 'a' + our_hash[1:]
2020-04-07T21:57:03.9136871Z     
2020-04-07T21:57:03.9137160Z         def mock_succesfull_upload_data_to_server(
2020-04-07T21:57:03.9137657Z                 url,  # pylint: disable=unused-argument
2020-04-07T21:57:03.9137956Z                 data,
2020-04-07T21:57:03.9138209Z                 timeout,
2020-04-07T21:57:03.9138471Z         ):
2020-04-07T21:57:03.9139004Z             # Can't compare data blobs as they are encrypted and as such can be
2020-04-07T21:57:03.9139310Z             # different each time
2020-04-07T21:57:03.9139764Z             assert 'data_blob' in data
2020-04-07T21:57:03.9140254Z             assert data['original_hash'] == our_hash
2020-04-07T21:57:03.9140750Z             assert data['last_modify_ts'] == last_write_ts
2020-04-07T21:57:03.9141228Z             assert 'index' in data
2020-04-07T21:57:03.9141761Z             assert len(data['data_blob']) == data['length']
2020-04-07T21:57:03.9142285Z             assert 'nonce' in data
2020-04-07T21:57:03.9142769Z             assert data['compression'] == 'zlib'
2020-04-07T21:57:03.9143035Z     
2020-04-07T21:57:03.9143301Z             assert timeout == ROTKEHLCHEN_SERVER_TIMEOUT
2020-04-07T21:57:03.9143790Z             return MockResponse(200, '{"success": true}')
2020-04-07T21:57:03.9144078Z     
2020-04-07T21:57:03.9144336Z         patched_put = patch.object(
2020-04-07T21:57:03.9144603Z             rotkehlchen_instance.premium.session,
2020-04-07T21:57:03.9145039Z             'put',
2020-04-07T21:57:03.9145328Z             side_effect=mock_succesfull_upload_data_to_server,
2020-04-07T21:57:03.9145598Z         )
2020-04-07T21:57:03.9145900Z         patched_get = create_patched_requests_get_for_premium(
2020-04-07T21:57:03.9146185Z             session=rotkehlchen_instance.premium.session,
2020-04-07T21:57:03.9146453Z             metadata_last_modify_ts=0,
2020-04-07T21:57:03.9146729Z             metadata_data_hash=remote_hash,
2020-04-07T21:57:03.9146994Z             # Smaller Remote DB size
2020-04-07T21:57:03.9147259Z             metadata_data_size=2,
2020-04-07T21:57:03.9147706Z             saved_data='foo',
2020-04-07T21:57:03.9147984Z         )
2020-04-07T21:57:03.9148235Z     
2020-04-07T21:57:03.9148495Z         with patched_get, patched_put:
2020-04-07T21:57:03.9148755Z             rotkehlchen_instance.premium_sync_manager.maybe_upload_data_to_server()
2020-04-07T21:57:03.9149030Z     
2020-04-07T21:57:03.9149285Z         now = ts_now()
2020-04-07T21:57:03.9149553Z         last_ts = rotkehlchen_instance.data.db.get_last_data_upload_ts()
2020-04-07T21:57:03.9150079Z         msg = 'The last data upload timestamp should have been saved in the db as now'
2020-04-07T21:57:03.9150583Z >       assert last_ts >= now and last_ts - now < 50, msg
2020-04-07T21:57:03.9150886Z E       AssertionError: The last data upload timestamp should have been saved in the db as now
2020-04-07T21:57:03.9151165Z E       assert (0 >= 1586296451)
2020-04-07T21:57:03.9151357Z 
2020-04-07T21:57:03.9151658Z rotkehlchen/tests/integration/test_premium.py:74: AssertionError
```

## Task

Figure out why and fix it

--- a/rotkehlchen/tests/integration/test_premium.py
+++ b/rotkehlchen/tests/integration/test_premium.py
@@ -65,10 +65,10 @@ def test_upload_data_to_server(rotkehlchen_instance, username, db_password):
         saved_data='foo',
     )
 
+    now = ts_now()
     with patched_get, patched_put:
         rotkehlchen_instance.premium_sync_manager.maybe_upload_data_to_server()
 
-    now = ts_now()
     last_ts = rotkehlchen_instance.data.db.get_last_data_upload_ts()
     msg = 'The last data upload timestamp should have been saved in the db as now'
     assert last_ts >= now and last_ts - now < 50, msg

